src/api/libcellml/analysermodel.h

```diff
@@ -461,6 +461,26 @@ class LIBCELLML_EXPORT AnalyserModel
      */
     bool needAcothFunction() const;
 
+    /**
+     * @brief Test to determine if @p variable1 and @p variable2 are the same or
+     * (directly or indirectly) equivalent.
+     *
+     * Test to see if @p variable1 is the same or (directly or indirectly)
+     * equivalent to @p variable2. Returns @c true if @p variable1 is the same
+     * or (directly or indirectly) equivalent to @p variable2 and @c false
+     * otherwise.
+     *
+     * @param variable1 The @c Variable to test if it is the same or (directly
+     * or indirectly) equivalent to @p variable2.
+     * @param variable2 The @c Variable that is potentially the same or
+     * (directly or indirectly) equivalent to @p variable1.
+     *
+     * @return @c true if @p variable1 is the same or (directly or indirectly)
+     * equivalent to @p variable2 and @c false otherwise.
+     */
+    bool isSameOrEquivalentVariable(const VariablePtr &variable1,
```

Comment:
convince me why this method should be checking "same" and "equivalent"...

Reply:
This PR is not fully ready to review and, indeed, I just refactored that method without changing its name. IOW, this method probably ought to be renamed `areEquivalentVariables()`? Also, it should take two `AnalyserVariablePtr`s as arguments, not two `VariablePtr`s.

Reply:
Partly done in commit 1bdb65338. About using passing two `AnalyserVariablePtr`s rather than two `VariablePtr`s, not sure it can be done in the end. Indeed, our generator relies on `AnalyserModel::areEquivalentVariables()` to retrieve the corresponding `AnalyserVariablePtr` of a `VariablePtr`...

However, I am going to have a quick go at what @kerimoyle described at https://github.com/cellml/libcellml/issues/762#issuecomment-716799564.

Reply:
Ok, from our discussion at today's libCellML meeting, it looks like this conversation can be considered as resolved.